quinn-proto/src/connection.rs

```diff
@@ -750,12 +750,32 @@ impl Connection {
         remote: SocketAddrV6,
         partial_decode: PartialDecode,
     ) -> Option<BytesMut> {
-        let crypto = if partial_decode.is_handshake() {
-            &self.handshake_crypto
+        let result = if partial_decode.is_handshake() {
+            partial_decode.finish(self.handshake_crypto.pn_decrypt_key())
+        } else if partial_decode.key_phase() == self.key_phase {
+            partial_decode.finish(self.crypto.as_ref().unwrap().pn_decrypt_key())
```

Comment:
Could a misbehaving peer cause us to panic here by sending a non-handshake packet before `crypto` is populated?